UnitySDK/Assets/ML-Agents/Scripts/RewardProvider/LowLevelRewardProvider.cs

```diff
@@ -0,0 +1,77 @@
+
+namespace MLAgents.RewardProvider
+{
+    /// <summary>
+    /// A legacy reward provider that can be used in an Agent as a way to easily upgrade
+    /// from the old reward system.
+    /// </summary>
+    public class LowLevelRewardProvider : IRewardProvider
+    {
+        float m_IncrementalReward;
+        float m_CumulativeReward;
+
+        public delegate void RewardReset(float reward);
+
+        public event RewardReset OnRewardProviderReset;
```

Comment:
This event is currently only consumed in the editor to draw the graph. I think this adds some complexity, I would argue to either remove the graph logging or put this section of the code between `#if UNITY_EDITOR` statements.

Reply:
I'd like to keep references to unity macros out of these classes, I have wrapped all of the uses of this in `#if UNITY_EDITOR`.

Reply:
Let me know if that works for you or if you want me to wrap this as well.